src/components/editContainer/editContainer.jsx

```diff
@@ -31,30 +33,44 @@ export class EditContainer extends Component {
         }
     }
 
-    escFunction (event) {
+    handleClick = (e) => {
+        if (this.containerRef.contains(e.target)) { //Click inside, don't do anything
+            return;
+        }
+        if (this.props.name !== '' && this.props.size !== 0 && this.props.size !== '') {
+            this.props.handleEnter()
+        } else if (this.props.name === '' && (this.props.size === '' || this.props.size === 0)) {
+            this.props.handleEsc()
+        }
+    }
+
+    escFunction = (event) => {
         if(event.keyCode === 27) {
             this.props.handleEsc()
         }
     }
 
     componentDidMount(){
         document.addEventListener("keydown", this.escFunction, false);
+        document.addEventListener("mousedown", this.handleClick, false);
     }
 
     componentWillUnmount(){
         document.removeEventListener("keydown", this.escFunction, false);
+        document.removeEventListener("mousedown", this.handleClick, false);
     }
   
     render () {
         return (
-            <div className="container">
+            <div className="container" ref={containerRef => this.containerRef = containerRef}>
                 <Grid container spacing={24}>
                     <Grid item xs={12}>
                         <Typography variant="headline" align="center">
                             <TextField
                                 inputRef={field => this.inputName = field}
                                 onKeyPress={this.handleNameKeyPress}
                                 onChange={this.props.handleNameChange}
+                                // onBlur={this.handleBlur}
```

Comment:
did you want to keep this commented out?

Reply:
Shoot, I'll remove those in a different PR